### PR TITLE
Cover/cmds/core/cmp: Add tests.

### PR DESCRIPTION
**Pull request by llogen**
Refactored cmp.go and added tests to cmp_test.go.

Signed-off-by: llogen <christophlange95@googlemail.com>

### CODE REVIEW

File: cmds/core/cmp/cmp.go [hunk 1]
```diff
@@ -56,12 +56,12 @@ func emit(rs io.ReadSeeker, c chan byte, offset int64) error {
 	}
 }
 
-func openFile(name string) (*os.File, error) {
+func openFile(stdin *os.File, name string) (*os.File, error) {
```
Comment (rminnich) [inline cmds/core/cmp/cmp.go:59]: The name openFile is a bit odd given that it is passed a file?

not sure if there is a better name? readFileOrStdin? not sure.

File: cmds/core/cmp/cmp.go [hunk 2]
```diff
@@ -70,53 +70,50 @@ func openFile(name string) (*os.File, error) {
 }
 
 // cmp is defined to fail with exit code 2
-func failf(format string, a ...interface{}) {
-	fmt.Fprintf(os.Stderr, format, a...)
-	os.Exit(2)
+func failf(w io.Writer, format string, a ...interface{}) int {
```
Comment (rminnich) [inline cmds/core/cmp/cmp.go:73]: could this not return a syscall.Error with value 2? i.e. the signature is fail(...) error and you return
return syscall.Errno(2)

what is 2 anyway?

File: cmds/core/cmp/cmp.go [hunk 3]
```diff
@@ -70,53 +70,50 @@ func openFile(name string) (*os.File, error) {
...
 	var err error
 
-	fnames := flag.Args()
-
 	cmpUnits := unit.DefaultUnits
 
 	off, err := unit.NewUnit(cmpUnits)
 	if err != nil {
-		failf("Could not create unit based on mapping: %v\n", err)
+		failf(w, "Could not create unit based on mapping: %v\n", err)
```
Comment (rminnich) [inline cmds/core/cmp/cmp.go:87]: ah, wait, fail is no longer failing! 

I would remove the fail functions, it's pretty useless, and cmp should change to return (int, error), and just have every line with fail do a return count, error

The caller can exit with os.Exit(2) if needed.

File: cmds/core/cmp/cmp.go (hunk 1 above)
Comment (llogen) [reply cmds/core/cmp/cmp.go:59]: yep sounds good

File: cmds/core/cmp/cmp.go (hunk 2 above)
Comment (llogen) [reply cmds/core/cmp/cmp.go:73]: There is a comment that says that cmp is defined to fail with exit code 2

**Review by rminnich — DISMISSED** [review]

**Review by walterchris — APPROVED** [review]

**Pull request merged**